src/main/java/net/blay09/mods/cookingforblockheads/item/ItemRecipeBook.java

```diff
@@ -1,107 +1,123 @@
-        }
-    }
-
-    @Override
-    public void getSubItems(Item item, CreativeTabs creativeTabs, List list) {
-        list.add(new ItemStack(item, 1, 3));
-        list.add(new ItemStack(item, 1, 0));
-        list.add(new ItemStack(item, 1, 1));
-    }
-
-    @Override
-    public IIcon getIconFromDamage(int damage) {
-        switch (damage) {
-            case TIER_2:
-                return iconTier2;
-            case TIER_0:
-                return iconTier0;
-            default:
-                return itemIcon;
-        }
-    }
-
-    @Override
-    public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer player) {
-        int guiId = GuiHandler.ITEM_RECIPE_BOOK;
-
-        player.openGui(
-                CookingForBlockheads.instance, guiId, world, (int) player.posX, (int) player.posY, (int) player.posZ);
-        return itemStack;
-    }
-
-    @Override
-    public void addInformation(ItemStack itemStack, EntityPlayer player, List list, boolean isShiftDown) {
-        super.addInformation(itemStack, player, list, isShiftDown);
-        switch (itemStack.getItemDamage()) {
-            case TIER_2:
-                list.add("\u00a7e" + I18n.format("cookingforblockheads:recipebook_tier2.tooltip"));
-                for (String s : I18n.format("cookingforblockheads:recipebook_tier2.tooltipDesc")
-                        .split("\\\\n")) {
-                    list.add("\u00a77" + s);
-                }
-                break;
-            case TIER_0:
-                list.add("\u00a7e" + I18n.format("cookingforblockheads:recipebook_tier0.tooltip"));
-                for (String s : I18n.format("cookingforblockheads:recipebook_tier0.tooltipDesc")
-                        .split("\\\\n")) {
-                    list.add("\u00a77" + s);
-                }
-                break;
-            case TIER_1:
-                list.add("\u00a7e" + I18n.format("cookingforblockheads:recipebook_tier1.tooltip"));
-                for (String s : I18n.format("cookingforblockheads:recipebook_tier1.tooltipDesc")
-                        .split("\\\\n")) {
-                    list.add("\u00a77" + s);
-                }
-                break;
-        }
-    }
-}
+package net.blay09.mods.cookingforblockheads.item;
+
+import java.util.List;
+import net.blay09.mods.cookingforblockheads.CookingForBlockheads;
+import net.blay09.mods.cookingforblockheads.GuiHandler;
+import net.minecraft.client.renderer.texture.IIconRegister;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+import net.minecraft.world.World;
+
+public class ItemRecipeBook extends Item {
+
+    public enum BookTiers {
+        TIER_0,
+        TIER_1,
+        TIER_2;
+
+        public static BookTiers from(int val) {

```

Comment:
This is _bad_, especially returning `null`. A better approach would be to use meaningful enum names corresponding to books names and associating a tier with the enum value.

Reply:
IMO, I don't see the value in changing to an enum at this time. I added C-style constants merely to give labels to otherwise naked ints, and was not prepared to retrofit in an enum construct into the existing code. 

I'm going to do some rebasing so this PR (or perhaps a new one, if github doesn't like history revision) only has the bug fix. Adding more will detract from the original PR intent.